### PR TITLE
re-refactor readOnes and update create and update for song

### PR DESCRIPTION
**Pull request by tjgersho**
re-refactor readones for Story, Song, Literacy-material and Film.
update create and update function for Song.

### CODE REVIEW

File: src/components/song/song.service.ts [hunk 1]
```diff
@@ -315,4 +353,50 @@ export class SongService {
...
+        node('song', 'Song', {
+          id: songId,
+          active: true,
+          owningOrgId: session.owningOrgId,
+        }),
+        relation('out', '', 'scriptureReferences'),
+        node('scriptureRanges', 'ScriptureRange', { active: true }),
+      ])
+      .with('collect(scriptureRanges) as items')
+      .return('items');
```
Comment (CarsonF) [inline src/components/song/song.service.ts:373]: There's no need to collect() & first() row just to run() it. 
```suggestion
      .return('scriptureRanges')
      .asResult<{ scriptureRanges: Node<{ start: number; end: number }> }>();
```
Then
```tsx
const results = await query.run();
return results.map(item => 
  verseToScriptureRange({
    start: item.properties.start,
    end: item.properties.end,
  })
);

File: src/components/song/song.service.ts [hunk 2]
```diff
@@ -162,31 +167,39 @@ export class SongService {
...
+
+      if (input.scriptureReferences) {
+        for (const sr of input.scriptureReferences) {
+          const verseRange = scriptureToVerseRange(sr);
+          query.create([
+            node('node'),
+            relation('out', '', 'scriptureReferences', { active: true }),
+            node('sr', 'ScriptureRange', {
+              start: verseRange.start,
+              end: verseRange.end,
```
Comment (CarsonF) [inline src/components/song/song.service.ts:188]:
```suggestion
              ...scriptureToVerseRange(sr),
```

File: src/components/literacy-material/literacy-material.service.ts [hunk 3]
```diff
@@ -315,52 +268,21 @@ export class LiteracyMaterialService {
...
-              start: verseRange.start,
-              end: verseRange.end,
-              active: true,
-              createdAt: DateTime.local(),
-            }),
-          ])
-          .return('lm')
-          .first();
-      }
+    if (input.scriptureReferences) {
```
Comment (CarsonF) [inline src/components/literacy-material/literacy-material.service.ts:271]: Can this check be moved to the service and allow nullable to be passed in?

File: src/components/literacy-material/literacy-material.service.ts [hunk 4]
```diff
@@ -291,21 +239,26 @@ export class LiteracyMaterialService {
...
 
-    const secured = parseSecuredProperties(result.propList, result.permList, {
-      name: true,
-    });
-
-    const scriptureReferences = await this.listScriptureReferences(
-      literacyMaterialId,
+    const scriptureReferences = await this.scriptureRefService.list(
+      id,
+      'LiteracyMaterial',
```
Comment (CarsonF) [inline src/components/literacy-material/literacy-material.service.ts:244]: Is this label really needed? Can't we just use the id with the 'BaseNode' label? Do we not index on `id` properties?

File: src/components/song/song.service.ts [hunk 5]
```diff
@@ -158,35 +160,36 @@ export class SongService {
...
 
-      const result = await query.first();
       if (!result) {
         throw new ServerException('failed to create a song');
       }
 
-      const id = result.id;
-
-      // add root admin to new song as an admin
-      await this.db.addRootAdminToBaseNodeAsAdmin(id, 'Song');
```
Comment (CarsonF) [inline src/components/song/song.service.ts:186]: This is the only one of the 4 that had this. Is this missing on all 4 now or is it not needed?

**Review by CarsonF — CHANGES_REQUESTED** [review]

File: src/components/song/song.service.ts (hunk 5 above)
Comment (tjgersho) [reply src/components/song/song.service.ts:186]: It is duplicated in `createBaseNode` function. So not needed.

File: src/components/literacy-material/literacy-material.service.ts (hunk 4 above)
Comment (tjgersho) [reply src/components/literacy-material/literacy-material.service.ts]: Yeah. right.

Comment (tjgersho) [reply src/components/literacy-material/literacy-material.service.ts]: Can use `BaseNode`.

File: src/components/literacy-material/literacy-material.service.ts (hunk 3 above)
Comment (tjgersho) [reply src/components/literacy-material/literacy-material.service.ts]: Sure

**Review by CarsonF — APPROVED** [review]
Thanks @tjgersho!

I just tweaked to use early returns and to pass session object

**Pull request merged**